src/base/static/components/form-fields/types/__tests__/add-attachment-button.test.js

```diff
@@ -0,0 +1,94 @@
+import React from "react";
+import { shallow } from "enzyme";
+const Util = require("../../../../js/utils.js");
+jest.mock("../../../../js/utils.js");
+
+import AddAttachmentButton from "../add-attachment-button";
+import FileField from "../file-field";
+import constants from "../../../../constants";
+
+describe("AddAttachmentButton", () => {
+  const defaultProps = {
+    id: "test-id",
+    name: "test-name",
+    label: "test-label",
+    onChange: () => {},
+    onAddAttachment: () => {},
+  };
+
+  test("renders add attachment button", () => {
+    const wrapper = shallow(<AddAttachmentButton {...defaultProps} />);
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  test("converts file to blob format", () => {
+    const onChangeProp = jest.fn();
+    const onAddAttachmentProp = jest.fn();
+    const props = {
+      ...defaultProps,
+      onChange: onChangeProp,
+      onAddAttachment: onAddAttachmentProp,
+    };
+    const wrapper = shallow(<AddAttachmentButton {...props} />);
+    const mockFile = new File(["My file contents"], "myFile");
+
+    const fileToCanvasMock = jest.fn((file, callback, options) => {
+      callback({
+        toBlob: canvasToBlobMock,
+        toDataURL: canvasToDataUrlMock,
+      });
+    });
+
+    Util.fileToCanvas.mockImplementation(fileToCanvasMock);
+
+    const canvasToBlobMock = jest.fn(cb => {
+      cb("blob");
+    });
+
+    const canvasToDataUrlMock = jest.fn(fileType => "data-url");
+
+    wrapper
+      .find(FileField)
+      .first()
+      .props()
+      .onChange({
+        persist: () => {},
+        target: {
+          name: "target-name",
+          files: [mockFile],
+        },
+      });
+
+    // Internally used functions
+    const toBlobCalls = canvasToBlobMock.mock.calls;
+    expect(toBlobCalls.length).toEqual(1);
+
+    const toDataUrlCalls = canvasToDataUrlMock.mock.calls;
+    expect(toDataUrlCalls.length).toEqual(1);
+    const toDataUrlCall = toDataUrlCalls[0];
+    expect(toDataUrlCall[0]).toEqual("image/jpeg");
+
+    const toCanvasCalls = fileToCanvasMock.mock.calls;
+    expect(toCanvasCalls.length).toEqual(1);
+    const toCanvasCall = toCanvasCalls[0];
+    expect(toCanvasCall[0]).toEqual(mockFile);
+    expect(toCanvasCall[2].maxHeight).toEqual(800);
+    expect(toCanvasCall[2].maxHeight).toEqual(800);
```

Comment:
should this be `maxWidth`?

Reply:
Jesus, good eye. I should use a different method to `expect` on these objects that isn't so brittle.